src/stdune/temp.ml

```diff
@@ -14,3 +14,18 @@ let create ~prefix ~suffix =
 let destroy fn =
   Path.unlink_no_err fn;
   tmp_files := Path.Set.remove !tmp_files fn
+
+let prng = lazy (Random.State.make_self_init ())
```

Comment:
This should share the same code path with `Temp.create`. We should also not add any additional at_exit callbacks.